### ISSUE TITLE
Question: How can we use this for odoo?

### ISSUE THREAD

**Issue by amkarthik** [acsone/kwkhtmltopdf#9]
Hi,
I tried to configure it to Odoo.
I run the server(using docker image), then created the report.url parameter to the server url.
I feels I am missing something.
Can we have detailed documentation here to configure with Odoo? (like wiki pages).
If I get to the steps to use it with Odoo, I can contribute to the documentation part.
Thanks

**Comment by sbidoul** [comment]
Hi, for Odoo the `report.url` does *not* need changing.

You need to
- set an environment variable `KWKHTMLTOPDF_SERVER_URL` before starting Odoo.
- put the kwkhtmltopdf client in place of wkhtmltopdf in your path, for example like [this](https://github.com/acsone/odoo-bedrock/blob/master/install/wkhtmltopdf.sh)

**Comment by amkarthik** [comment]
@sbidoul 
I did as you said.
Getting the following error on odoo and kwkhtmltopdf_server.
On odoo:
 Wkhtmltopdf failed (error code: 255). Message: b'kwkhtmltopdf server error, consult server log\n' 

On server:
2019/12/31 10:58:52 POST /
2019/12/31 10:58:52 [--disable-local-file-access --cookie session_id ef1c7335b1d617adf115ae92cc68b135eabf0b9e --quiet --page-size Letter --margin-top 40.0 --dpi 90 --zoom 1.0666666666666667 --header-spacing 35 --margin-left 7.0 --margin-bottom 25.0 --margin-right 7.0 --orientation Portrait --header-html /tmp/kwk584382483/report.header.tmp.jcs73mgu.html --footer-html /tmp/kwk584382483/report.footer.tmp.jot_q1fy.html /tmp/kwk584382483/report.body.tmp.0.t7wqeq93.html -] starting
Exit with code 1 due to network error: ConnectionRefusedError
2019/12/31 10:58:52 exit status 1

**Comment by sbidoul** [comment]
You probably have the kwkhtmltopdf server on the same machine as Odoo, trying to call `wkhtmltopdf`, which is now the kwkhtmltopdf client, trying to call the server again (with a wrong URL).

Try setting the `KWKHTMLTOPDF_BIN` env var to the full path of the original `wkhtmltopdf` executable.

**Comment by amkarthik** [comment]
Yes. I am running kwkhtmltopdf server on the same machine using docker image.
I set the env variable KWKHTMLTOPDF_BIN as you said. Even I get the same error.
Help me to solve this.
Thanks

**Comment by amkarthik** [comment]
@sbidoul 
I figured out the issue in my case.
I run my odoo docker image using the below command 
docker run --network=host -it -v opt/my_docker/tests/odoo-dev/project/odoo-12/:/opt/odoo/odoo-12 
i.e i told it to use the host network (make it to use Postgres running on the host machine)
When printing the pdf from odoo, it stores the tmp headers, footers, body html files inside the odoo container not on the host machine, the assert url in the html files hold the value of the host machine.
So, kwkhtmltopdf server can't find the html files on the host machine so it raised the above error.

So I solved it by mounting the host /tmp with the odoo docker container /tmp. Like below
docker run --network=host -it -v opt/my_docker/tests/odoo-dev/project/odoo-12/:/opt/odoo/odoo-12 -v /tmp:/tmp
It solved the issue and worked like a charm.

I hope I have to find a better workflow for my docker development environment that suits my need.
Thanks.
